### ISSUE TITLE
TempoChange related crash

### ISSUE THREAD

**Issue by izzyreal** [izzyreal/mpc#33]
Reported by Eagle Andrii. Often happens when he "records track 2 before 3 when I press record + play"
```
Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000060

VM Regions Near 0x60:
--> 
    __TEXT                 000000010657d000-0000000106c44000 [ 6940K] r-x/rwx SM=COW  /Applications/vmpc.app/Contents/MacOS/vmpc

Thread 0 Crashed:: JUCE Message Thread  Dispatch queue: com.apple.main-thread
0   nl.izmar.vmpc                 	0x000000010676387b mpc::sequencer::Sequence::getTempoChangeEvents() + 43
1   nl.izmar.vmpc                 	0x000000010681d122 mpc::ui::sequencer::SequencerObserver::displayTempoLabel() + 66
2   nl.izmar.vmpc                 	0x000000010681bb95 mpc::ui::sequencer::SequencerObserver::displayTempo() + 21
3   nl.izmar.vmpc                 	0x000000010681ddf9 mpc::ui::sequencer::SequencerObserver::update(moduru::observer::Observable*, nonstd::any_lite::any) + 2361
4   nl.izmar.vmpc                 	0x0000000106589573 moduru::observer::Observable::notifyObservers(nonstd::any_lite::any) + 99
5   nl.izmar.vmpc                 	0x000000010676d489 mpc::sequencer::Sequencer::move(int) + 1081
6   nl.izmar.vmpc                 	0x000000010676c576 mpc::sequencer::Sequencer::play(bool) + 342
7   nl.izmar.vmpc                 	0x000000010666956e mpc::controls::BaseControls::play() + 910
8   nl.izmar.vmpc                 	0x000000010670d7f7 mpc::hardware::Button::push() + 1335
9   nl.izmar.vmpc                 	0x00000001068784dc InputCatcherControl::keyPressed(juce::KeyPress const&) + 3388
10  nl.izmar.vmpc                 	0x00000001069e7100 juce::ComponentPeer::handleKeyPress(juce::KeyPress const&) + 272
```

**Comment by izzyreal** [comment]
Steps to reproduce:
```
1. Open vMPC
2. Load my All_SQNG1
3. Load All_PGMS .APS
4. Rec track-01
5.Rec track-02
6.Rec +play 
7 errors
```

**Comment by izzyreal** [comment]
[A.zip](https://github.com/izzyreal/mpc/files/4515682/A.zip)
This has files needed to reproduce the crash.

**Comment by izzyreal** [comment]
Resolved in 2e9d18b66f90e059a2976af9446801c181211868
